### ISSUE TITLE
SAM - AWS::Serverless::Api - Property Domain is not allowed

### ISSUE THREAD

**Issue by ChaseIngebritson** [aws/aws-toolkit-vscode#2467]
**Describe the bug**

When including the Domain property in a AWS::Serverless::Api resource, I'm given the error `Property Domain is not allowed.`. The  Domain property in the [SAM specification documentation](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/sam-resource-api.html#sam-api-domain) seems to show that it should be allowed.

**To Reproduce**

Generate a sam template and include a AWS::Serverless::Api resource along the lines of:

```yaml
MyApi:
  Type: AWS::Serverless::Api
  Properties:
    StageName: STAGE
    Domain:
      DomainName: DOMAIN
      CertificateArn: ARN
```
An error will be displayed stating that the property is not allowed:

<img width="1047" alt="Screen Shot 2022-02-11 at 9 01 04 PM" src="https://user-images.githubusercontent.com/11217065/153694197-c15b0849-60ea-4171-8d85-c58cb512266c.png">

**Expected behavior**

Given the SAM documentation, this seems like this property should be allowed.

**Desktop (please complete the following information):**

-   OS: MacOS 12.2.1
-   Visual Studio Code Version: 1.64.2
-   AWS Toolkit Version: 1.35.0

**Comment by rubenfonseca** [comment]
Hi @ChaseIngebritson thank you for the very good and detailed report. We merged a PR on goformation that should fix the problem. Thank you again!

🎉 This PR is included in goformation version 6.0.11 🎉

The release is available on [GitHub release](https://github.com/awslabs/goformation/releases/tag/v6.0.11)

**Comment by jpinkney-aws** [comment]
As @rubenfonseca mentioned, the schema in the goformation package is now updated and you should no longer see any errors with Domain 🎉.

_Note:_ You might need to refresh VSCode to pick up the new schema
